lib/common/memory/src/mmap_ops.rs

```diff
@@ -39,22 +38,15 @@ pub fn create_and_ensure_length(path: &Path, length: usize) -> io::Result<File>
 
         std::fs::rename(&temp_path, path)?;
 
-        OpenOptions::new()
-            .read(true)
-            .write(true)
-            .create(false)
-            .truncate(false)
-            .open(path)
+        OpenOptions::new().read(true).write(true).open(path)
     }
 }
 
 pub fn open_read_mmap(path: &Path, advice: AdviceSetting, populate: bool) -> io::Result<Mmap> {
     let file = OpenOptions::new()
         .read(true)
-        .write(false)
         .append(true)
         .create(true)
-        .truncate(false)
         .open(path)?;
```

Comment:
Any specific reason why we need `append` and `create` here?

Reply:
I _think_ it was copy pasted from the read/write implementation and we forgot to move these options.

The [documentation](https://docs.rs/memmap2/latest/memmap2/struct.Mmap.html#method.map) makes me think that only having the read flag is good enough.

I cannot think of another reason for us needing these two mutability flags.

Reply:
Based on the above I think we should remove the `append` and `create` options. Then we can merge!